Hikari9/comfortable-swipe
# Use comfortable-swipe as systemd service

This is intended to address issues #7 and #13.

# Summary

This pull request adds `comfortable-swipe` as a systemd service.

This is preferred over a local startup application since touchpad gestures are semantically fitting to be called a service. Furthermore, the methods such as start and stop are standard commands usually found in services.

This also bypasses the input device permissions required by `libinput-debug-gestures`, since service daemons would already be handled as a superuser.

# How to Test

1. Have a fresh clone of this branch:
    ```bash
    git clone https://github.com/Hikari9/comfortable-swipe-ubuntu --branch feature-systemd comfortable-swipe-feature-systemd
    cd comfortable-swipe-feature-systemd
    ```
2. Run
    ```bash
    sudo apt-get install libinput-tools libxdo-dev
    bash install
    ```
3. Check if swiping works:
    - Edit `/usr/local/share/comfortable-swipe.conf`
    - Configure keyboard mapping to fit your needs (e.g. `ctrl+alt+Right` for swipe left 3-fingers)
    - Refer to [README.md](https://github.com/Hikari9/comfortable-swipe-ubuntu/blob/8b9b9601549a22c77019831941c64c8f5b9a25ae/README.md) for config list
4. Reboot and check if swiping still works.
5. Comment any problems. Please make sure to add the following to your comment:
    - Content of `.log` (found in the cloned folder)
    - Output of `comfortable-swipe status`
